src/mono/mono/mini/mini-trampolines.c

```diff
@@ -632,11 +632,13 @@ common_call_trampoline (host_mgreg_t *regs, guint8 *code, MonoMethod *m, MonoVTa
 		return NULL;
 
 	if (generic_virtual || variant_iface) {
-		if (m_class_is_valuetype (vt->klass)) /*FIXME is this required variant iface?*/
+		if (m_class_is_valuetype (vt->klass)  && !mini_method_is_default_method (m)) /*FIXME is this required variant iface?*/
 			need_unbox_tramp = TRUE;
 	} else if (orig_vtable_slot) {
-		if (m_class_is_valuetype (m->klass))
+		if (m_class_is_valuetype (m->klass)) {
+			g_assert (!mini_method_is_default_method (m));
```

Comment:
Why don't you add the same condition as above?

Reply:
Good question.
I couldn't tell if we could ever end up in this branch with a valuetype klass and a default interface method.

I thought it would be better to add an assertion (because if it does ever happen the failure will be right at this spot) instead of skipping the `need_unbox_tramp`.  If we erroneous add or don't add an unbox trampoline, we will get a crash later when we actually call the underlying managed method and it uses `this` in some way.  That will be a crash in some unrelated managed code and it won't be obvious how to trace it back to this trampoline.  With the assert if anyone comes up with a way to violate it the assumptions it will break in the place where we need to fix it.

Reply:
That's right. The assertion would definitely make debugging easier. But my confusion is that in the future, when someone hit this assertion, I feel the solution would be to skip adding unbox trampoline. Or you think more work would be needed on top of that?

Reply:
> That's right. The assertion would definitely make debugging easier. But my confusion is that in the future, when someone hit this assertion, I feel the solution would be to skip adding unbox trampoline. Or you think more work would be needed on top of that?

I don't know. I can't tell from the code how we end up in this branch with a DIM. I'd need to see an example and I couldn't make one.

And also I can't tell if in that case the right thing would be to add an unbox, or to not add one. To be honest I was hoping one of our existing tests  would break on the assert, and then I'd know that things were fine as is. 

But nothing broke. So I still don't know whether that case can happen. And what we should do if it does happen.

Reply:
make sense!